inc/classes/optimization/class-cache-dynamic-resource.php

```diff
@@ -167,7 +167,7 @@ public function is_allowed() {
 	public function is_excluded_file( $src ) {
 		$file = get_rocket_parse_url( $src );
 
-		if ( ! preg_match( '#\.php$#', $file['path'] ) ) {
+		if ( is_array( $file ) && ! empty( $file['path'] ) && ! preg_match( '#\.php$#', $file['path'] ) ) {
```

Comment:
`isset( $file['path'] )` does the following checks for us:

- `$file` is an array <- inferred because we are checking for a key.
- Key `'path'` is defined
- `$file['path'` is not `null`